concrete/single_pages/dashboard/system/registration/open.php

```diff
@@ -86,14 +86,37 @@
             </label>
         </div>
     </div>
-
+    <div class="form-group">
+        <label class="control-label"><?php echo t('Registration form') ?></label>
+        <div class="checkbox">
+            <label>
+                <input type="checkbox" name="display_username_field" value="1"
+                       style="" <?php echo ($display_username_field) ? 'checked' : '' ?> />
+                <span><?php echo t('Ask for Username on registration form') ?></span>
+            </label>
+        </div>
+        <div class="checkbox">
+            <label>
+                <input type="checkbox" name="display_confirm_password_field" value="1"
+                       style="" <?php echo ($display_confirm_password_field) ? 'checked' : '' ?> />
+                <span><?php echo t('Ask to confirm password on registration form') ?></span>
+            </label>
+        </div>
+    </div>
     <div class="ccm-dashboard-form-actions-wrapper">
         <div class="ccm-dashboard-form-actions">
             <?= $h->submit(t('Save'), 'registration-type-form', 'right', 'btn-primary'); ?>
         </div>
     </div>
 </form>
 
+<div id="dialog-confirm" style="display: none" title="<?= t('Do you want to apply?') ?>">
+    <p><?=t('You have to disable ask for Username on login form, if you want to disable it.') ?></p>
```

Comment:
"You have to disable ask for Username on login form, if you want to disable it." does not make sense to me. Can you explain what you mean by this? The javascript looks like you force `email_as_username` to turn on if `display_username_field` gets disabled?

Reply:
The reason is here-
https://github.com/concrete5/concrete5/blob/develop/concrete/controllers/single_page/register.php#L200-L205

If register form doesn't have `uName` field but `concrete.user.registration.email_registration ` is `false` it's try to login with `uName`, which we don't have. And then it returns error.

We could set auto generated `uName` on the controller, but I think when the site owner will disable `uName` field on register form, he may wants to login user by `uEmail`.

About the message, I hope you may suggest a better message which make sense. Me also, not satisfied with it.